### PR TITLE
Setup project with FastAPI

### PR DESCRIPTION
**Pull request by Haider8**

### CODE REVIEW

**Comment by Haider8** [comment]
@iamareebjamal Please review

**Comment by iamareebjamal** [comment]
Just fox hound errors. Rest looks fine

File: app/db/base.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Import all the models, so that Base has them before being
+# imported by Alembic
+from app.db.base_class import Base
+from app.db_models.user import User
```
[the remark this answers is not in the document]
Comment (Haider8) [reply app/db/base.py:4]: Ignore as these are getting used in alembic

File: app/crud/user.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+
+from sqlalchemy.orm import Session
+
+from app.core.security import get_password_hash, verify_password
```
[the remark this answers is not in the document]
Comment (Haider8) [reply app/crud/user.py:5]: Use at line 45

File: alembic/versions/89255db472dd_add_user_table.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ def upgrade():
     sa.Column('id', sa.Integer(), nullable=False),
     sa.Column('username', sa.String(length=64), nullable=True),
     sa.Column('email', sa.String(length=120), nullable=True),
-    sa.Column('password_hash', sa.String(length=128), nullable=True),
+    sa.Column('password_hash', sa.String(), nullable=True),
```
[the remark this answers is not in the document]
Comment (Haider8) [reply alembic/versions/89255db472dd_add_user_table.py:25]: Automatically generated by alembic. Ignore

**Comment by Haider8** [comment]
@iamareebjamal Please review

**Pull request merged**